### PR TITLE
VSTS-344 VSTS-345 Show issues fixed in PR and accepted issues in build summary when possible

### PR DESCRIPTION
**Pull request by 7PH**
This PR tries and fetch measures in the list `[new_violations, pull_request_fixed_issues, new_accepted_issues, new_security_hotspots, new_coverage, new_duplicated_lines_density]` respecting this logic:
- Doesn't try to show `pull_request_fixed_issues` when not on pull requests
- Doesn't show coverage or duplication data if there's none to show
- Doesn't try to fetch measures if there are not listed in available metrics (`/api/metrics/search`)
- If the metric exists but could not get the measure (possible, see next paragraph), tries to get the measure value from the QG condition
- Gracefully degrades the UI when some measures are not available (see examples)

Known limitation: Measures will only be able to be fetched when a user token is used for the analysis. Project analysis and Global analysis tokens don't have the `Browser project` permission necessary to get the project measures.

This PR also includes a fix after SSF-194 which was double-stringifying scanner params in tl variables

Examples from validation
- All measures available, on pull request
image
- No coverage or duplication data, on branch
image
- Failing condition
image
- SQ <= 10.3, no coverage and dup data
image
- SQ LTS
image
- SC, failed
image
- SC, passed (they don't have `new_violations` metric so it won't show up until they move to the new CCT)
image

### CODE REVIEW

File: commonv5/ts/__tests__/publish-task-test.ts [hunk 1]
```diff
@@ -317,3 +341,122 @@ it("task should not fail the task even if all ceTasks timeout", async () => {
...
+    // spy on publishBuildSummary
+    expect(serverUtils.publishBuildSummary).toHaveBeenCalledTimes(1);
+    const buildSummary = (serverUtils.publishBuildSummary as any).mock.calls[0][0];
+    expect(buildSummary).toMatch("Quality Gate passed (componentName)");
+  });
+
+  it.each([
+    [EndpointType.SonarQube, SQ_ENDPOINT, { "sonar.pullrequest.key": "123" }, true],
+    [EndpointType.SonarQube, SQ_ENDPOINT, {}, false],
+    [EndpointType.SonarQube, SQ_ENDPOINT, { "sonar.branch.name": "some-branch" }, false],
```
Comment (stanislavhh) [inline commonv5/ts/__tests__/publish-task-test.ts:431]: Are these 2 cases relevant? can we show fixed issues if there is no pr key?

**Review by stanislavhh — APPROVED** [review]
Awesome work 🎉 
Code is clean and coverage is high ❤️ 

Left some tiny comments/questions and also I see an error when running pipeline on main branch with project analysis token
image

File: commonv5/ts/helpers/api.ts [hunk 2]
```diff
@@ -47,3 +56,26 @@ export async function fetchMetrics(
...
+      true,
+      data,
+    )) as MeasureResponse;
+    return response.component.measures;
+  } catch (error) {
+    if (error?.message) {
+      tl.error(error.message);
+    } else if (error) {
+      tl.error(JSON.stringify(error));
+    }
```
[the remark this answers is not in the document]
Comment (7PH) [reply commonv5/ts/helpers/api.ts:77]: It's actually showing the error that you posted above 🤔 We don't want that as we want to silence this error, so I'm dropping it specifically for `fetchComponentMeasures`, unless you have concerns

File: commonv5/ts/__tests__/publish-task-test.ts (hunk 1 above)
Comment (7PH) [reply commonv5/ts/__tests__/publish-task-test.ts:431]: You're right, if there's no PR key, the measure can't be fetched on SQ anyway, so I've dropped it, thanks

File: commonv5/ts/helpers/api.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (7PH) [reply commonv5/ts/helpers/api.ts]: I just validated that dropping `tl.error` stops showing the red error message 👍

**Pull request merged**